src/mapblock_mesh.cpp

```diff
@@ -283,20 +319,62 @@ static u16 getSmoothLightCombined(const v3s16 &p, MeshMakeData *data)
 /*
 	Calculate smooth lighting at the given corner of p.
 	Both light banks.
+	Node at p is solid, and thus the lighting is face-dependent.
+*/
+u16 getSmoothLightSolid(const v3s16 &p, const v3s16 &face_dir, const v3s16 &corner, MeshMakeData *data)
+{
+	v3s16 neighbor_offset1, neighbor_offset2;
+
+	if (face_dir.X != 0) {
+		neighbor_offset1 = v3s16(0, corner.Y, 0);
+		neighbor_offset2 = v3s16(0, 0, corner.Z);
+	} else if (face_dir.Y != 0) {
+		neighbor_offset1 = v3s16(0, 0, corner.Z);
+		neighbor_offset2 = v3s16(corner.X, 0, 0);
+	} else if (face_dir.Z != 0) {
+		neighbor_offset1 = v3s16(corner.X,0,0);
+		neighbor_offset2 = v3s16(0,corner.Y,0);
+	}
```

Comment:
Seems that you define a coordinate system here. That should be described as it’s not obvious that `face_dir`, `neighbor_offset1` and `neighbor_offset2` form the basis, neither that `corner` is their sum.

Reply:
Will do